### PR TITLE
Detect external hook managers and warn during enable

### LINKED ISSUE #228: entire enable hooks ignored when core.hooksPath is overridden (Husky)
`entire enable` installs hooks in `.git/hooks/`, but Husky sets `core.hooksPath=.husky/_`. Git never runs Entire's hooks, so commits have no `Entire-Checkpoint` trailer.

**Repro:** Enable Entire in a Husky project, commit normally, check `git log -1` — no trailer.

**Workaround:** `git -c core.hooksPath=.git/hooks commit -m "message"`

**Fix:** `entire enable` should read `core.hooksPath` and install there (or chain from existing hooks).

Entire CLI 0.4.2, macOS, Husky via `.husky/_`

### PR DESCRIPTION
**Pull request by squishykid**
## Summary

- Detect Husky, Lefthook, pre-commit, and Overcommit during `entire enable` via filesystem checks (no file reads)
- Husky users get copy-paste instructions for durable hook integration (since Husky overwrites hooks on `npm install`)
- Other hook managers get a note to re-run `entire enable` if hooks are overwritten
- Lefthook detection covers all 16 valid config filenames (`{.,}lefthook{,-local}.{yml,yaml,json,toml}`)

## Test plan

- [x] `mise run fmt && mise run lint && mise run test:ci` passes
- [x] 17 unit tests covering all managers, dedup, warning output, edge cases
- [x] Manual test: temp repo with `.husky/_/` → warning with 4 hook commands shown
- [x] Manual test: temp repo with `lefthook.yml` → note shown
- [x] Manual test: clean repo → no warning

closes #228

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: cmd/entire/cli/strategy/hook_managers.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+// detectHookManagers checks the repository root for known hook manager config
+// files/directories. Detection is filesystem-only (os.Stat, no file reads).
+func detectHookManagers(repoRoot string) []hookManager {
+	var managers []hookManager
+
+	checks := []hookManager{
+		{"Husky", ".husky/", true},
+		{"pre-commit", ".pre-commit-config.yaml", false},
+		{"Overcommit", ".overcommit.yml", false},
```
[the remark this answers is not in the document]
Comment (squishykid) [reply cmd/entire/cli/strategy/hook_managers.go:28]: Presence of .husky/_ dir means that .husky/ dir also exists. This is an acceptable false positive.

**Review by Soph — APPROVED** [review]
